src/transformers/trainer_utils.py

```diff
@@ -785,3 +785,32 @@ def _remove_columns(self, feature: dict) -> dict:
     def __call__(self, features: List[dict]):
         features = [self._remove_columns(feature) for feature in features]
         return self.data_collator(features)
+
+
+def check_target_module_exists(optim_target_modules, key: str):
```

Comment:
I would be nice to have a test for this to make sure we're picking up the right modules

Reply:
Note that this is adapted from PEFT, which has a bunch of tests: https://github.com/huggingface/peft/blob/6008f272a565f56c146c5d9fd78d00cb24392d7b/tests/test_tuners_utils.py#L157

However, it's not 100% identical, so copying over the tests that apply here could still be useful.

Reply:
ok I will add tests for this !

Reply:
Done !